### PR TITLE
Fix `metric_tensor` with non-standard operations

### LINKED ISSUE #3576: [BUG] `metric_tensor` does not take `qml.adjoint` correctly into account
### Expected behavior

The transform `metric_tensor` respects the `qml.adjoint` transform.

### Actual behavior

In the following example, the `adjoint` of one particular operation in the circuit is realized in three ways:
1. Via classical preprocessing of the rotation angle, corresponding to inversion.
2. Via the deprecated in-place inversion, yielding the same metric tensor as 1.
3. Via `qml.adjoint`, yielding a differing metric tensor.

```python
dev = qml.device("default.qubit", wires=3)

@qml.qnode(dev, interface="jax")
def circuit(x):
    qml.RX(x[0], wires=0)
    qml.CNOT([0, 1])
    qml.RZ(0.4, wires=1)
    qml.CNOT([0, 1])
    qml.RX(x[0], wires=0)
#     qml.RZ(-x[1], wires=0) # Produces the correct MT via preprocessing
#     qml.RZ(x[1], wires=0).inv() # Produces the correct MT
#     qml.adjoint(qml.RZ)(x[1], wires=0) # Produces the wrong MT
    return qml.expval(qml.PauliZ(0))

x = pnp.array([0.543, 0.352], requires_grad = True)
y = pnp.array(-0.654, requires_grad = True)

print(qml.transforms.metric_tensor(circuit)(x))
```

### Additional information

_No response_

### Source code

_No response_

### Tracebacks

_No response_

### System information

```shell
Name: PennyLane
Version: 0.28.0.dev0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /home/david/venvs/xenv/lib/python3.10/site-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, requests, retworkx, scipy, semantic-version, toml
Required-by: PennyLane-Lightning

Platform info:           Linux-5.15.0-56-generic-x86_64-with-glibc2.35
Python version:          3.10.6
Numpy version:           1.23.4
Scipy version:           1.9.3
Installed devices:
- default.gaussian (PennyLane-0.28.0.dev0)
- default.mixed (PennyLane-0.28.0.dev0)
- default.qubit (PennyLane-0.28.0.dev0)
- default.qubit.autograd (PennyLane-0.28.0.dev0)
- default.qubit.jax (PennyLane-0.28.0.dev0)
- default.qubit.tf (PennyLane-0.28.0.dev0)
- default.qubit.torch (PennyLane-0.28.0.dev0)
- default.qutrit (PennyLane-0.28.0.dev0)
- null.qubit (PennyLane-0.28.0.dev0)
- lightning.qubit (PennyLane-Lightning-0.27.0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

### PR DESCRIPTION
**Pull request by dwierichs**
**Context:**
The metric tensor for `Adjoint` operations is incorrect, see the linked issue.
This is caused by two different modes in which `qml.generator` is used within `qml.metric_tensor`.
This difference caused some parts to assume that the prefactor of the generator is contained in the applied gates
within the generated tapes, whereas other parts assumed that these prefactors still need to be taken into account in postprocessing. 

**Description of the Change:**
The usage of `qml.generator` is unified to the `"prefactor"` format, separating prefactors out into postprocessing.

**Benefits:**
`metric_tensor` produces correct results for non-standard ops, including `Adjoint`.

**Possible Drawbacks:**
N/A

**Related GitHub Issues:**
closes #3576

### CODE REVIEW

**Comment by josh146** [comment]
Nice catch @dwierichs! Thinking out loud here --- should we remove the 'legacy' modes from `qml.generator`?

**Comment by dwierichs** [comment]
>should we remove the 'legacy' modes from qml.generator? 

I'm not sure. Within the codebase, `qml.generator` is not used in a lot of places, but both `format="observable"` and the default `format="prefactor"` are used.

File: pennylane/transforms/metric_tensor.py [hunk 1]
```diff
@@ -567,6 +567,7 @@ def _metric_tensor_hadamard(tape, allow_nonunitary, aux_wire, device_wires):
         list[pennylane.tape.QuantumTape]: Tapes to evaluate the metric tensor
         callable: processing function to obtain the metric tensor from the tape results
     """
+    print(tape.operations)
```
Comment (albi3ro) [inline pennylane/transforms/metric_tensor.py:570]:
```suggestion
```

File: pennylane/transforms/metric_tensor.py [hunk 2]
```diff
@@ -479,7 +479,7 @@ def _get_gen_op(op, allow_nonunitary, aux_wire):
 
     except KeyError as e:
         if allow_nonunitary:
-            mat = qml.matrix(qml.generator(op, format="observable"))
+            mat = qml.matrix(qml.generator(op)[0])
             return qml.ControlledQubitUnitary(mat, control_wires=aux_wire, wires=op.wires)
```
Comment (albi3ro) [inline pennylane/transforms/metric_tensor.py:483]: Why not just `qml.ctrl(qml.generator(op)[0], control=aux_wire)`?

Comment (dwierichs) [reply pennylane/transforms/metric_tensor.py:483]: Indeed! :100: This is a legacy line.

**Review by albi3ro — APPROVED** [review]
I'm going to go ahead and approve, especially since its a bug fix and I want these updated tests in for the push to remove in-place inversion.

File: pennylane/transforms/metric_tensor.py (hunk 2 above)
Comment (dwierichs) [reply pennylane/transforms/metric_tensor.py:483]: In order to realize this, the `generator` of operations needs to be a queued object, in order for `ctrl` to lead to queuing as well. This change to the generators is left for a separate PR.
